### DISCUSSION TITLE
How do you get the character clicked on?

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [unoplatform/uno#13415] [General]
In WinUI 3 you can use the TextPointer.GetCharacterRect method to build an array of hit boxes for the text in a TextBlock.
A RichTextBlock has built-in functionality to do this.

But Uno doesn't appear to support any way to get the character clicked on in displayed text?  

Is there some way I don't know about?  This is ESSENTIAL to my application.  

Mark

**Comment by FrozDark** [comment]
You may able to implement this with https://github.com/toptensoftware/RichTextKit as I did on Xamarin.Forms

**Reply by MarkBrainfire** [reply]
Thank for your reply FrozDark. 

This is amazing to me.  After all these years Uno Platform can't tell what text a user clicks on?  Unbelievable!

**Reply by jeromelaban** [reply]
Thanks for the report! 

This is indeed not implemented at this time. Uno is progressing as per the requests from the community and clients that are driving the backlog. If a feature is not there yet, it means that it's not been requested much or is a part of a much larger change that has not been prioritized yet.

If you can open an issue with a sample of what you're trying to do, it will help prioritizing it.

**Reply by limeniye** [reply]
Hello, @MarkBrainfire.
Did you tried the [IsTextSelectionEnabled](https://learn.microsoft.com/en-us/dotnet/api/communitytoolkit.winui.ui.controls.markdown.render.markdownrenderer.istextselectionenabled?view=win-comm-toolkit-dotnet-7.0#communitytoolkit-winui-ui-controls-markdown-render-markdownrenderer-istextselectionenabled) for [MarkdownTextBlock](https://github.com/MicrosoftDocs/WindowsCommunityToolkitDocs/blob/main/docs/controls/MarkdownTextBlock.md)? Is that good alternative of your expectations?

```C#
private void MarkdownText_CodeBlockResolving(object sender, CodeBlockResolvingEventArgs e)
{
    if (e.CodeLanguage == "CUSTOM")
    {
        e.Handled = true;
        e.InlineCollection.Add(new Run { Foreground = new SolidColorBrush(Colors.Red), Text = e.Text, FontWeight = FontWeights.Bold });
    }
}
```
